key.core/src/main/java/de/uka/ilkd/key/proof/mgt/SpecificationRepository.java

```diff
@@ -1847,4 +1850,87 @@ public ImmutableSet<WellDefinednessCheck> getAllWdChecks() {
         }
         return result;
     }
+
+
+    // region Support SetStatement and JmlAssert
+    private final Map<Statement, JmlStatementSpec> statementMap = new IdentityHashMap<>();
+
+    @Nullable
+    public JmlStatementSpec getStatementSpec(Statement statement) {
+        return statementMap.get(statement);
+    }
+
+    public JmlStatementSpec addStatementSpec(Statement statement, JmlStatementSpec spec) {
+        return statementMap.put(statement, spec);
+    }
```

Comment:
Is it always ensured that there are different assertions/set statements for different branches? When unrolling, inling etc.?

Reply:
You/we need to ensure, that a new `SetStatement` is created. 

Something that needs to be assured by `ProgVarReplacer`, and was not done before.